daemon/algod/server.go

```diff
@@ -171,13 +171,25 @@ func (s *Server) Initialize(cfg config.Local, phonebookAddresses []string, genes
 			NodeExporterPath:          cfg.NodeExporterPath,
 		})
 
-	s.node, err = node.MakeFull(s.log, s.RootPath, cfg, phonebookAddresses, s.Genesis)
+	if cfg.NodeSyncMode {
+		dataNode, err := node.MakeData(s.log, s.RootPath, cfg, phonebookAddresses, s.Genesis)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("node has not been installed: %s", err)
+		}
+		if err != nil {
+			return fmt.Errorf("couldn't initialize the node: %s", err)
+		}
+		s.node = apiServer.DataNode{AlgorandDataNode: dataNode}
+		return nil
+	}
+	fullNode, err := node.MakeFull(s.log, s.RootPath, cfg, phonebookAddresses, s.Genesis)
 	if os.IsNotExist(err) {
 		return fmt.Errorf("node has not been installed: %s", err)
 	}
 	if err != nil {
 		return fmt.Errorf("couldn't initialize the node: %s", err)
 	}
+	s.node = apiServer.APINode{AlgorandFullNode: fullNode}
```

Comment:
I need to inspect the new interfaces more closely, I wonder if there's a way to organize the new interfaces that could consolidate the duplicate error handling.

If not, consider putting this block of code in a function that returns the node object.